hugr-py/src/hugr/tys.py

```diff
@@ -103,6 +116,10 @@ def _to_serial(self) -> stys.TypeTypeParam:
     def __str__(self) -> str:
         return str(self.bound)
 
+    def to_model(self) -> model.Term:
+        # Note that we drop the bound.
```

Comment:
Why are we dropping the bounds here and in `BoundedNatParam`?

Reply:
They have no representation in hugr-model. There was a big discussion on this half a year ago, resulting in the realisation that the bounds introduce subtyping and therefore make the type system unnecessarily complex. But without having a stable serialisation format, we were unable to change anything about that, leading to an in part still unresolved chicken egg problem. Bounds for type variables in some situations are emulated with constraints.